src/daft-table/src/ops/joins/mod.rs

```diff
@@ -121,6 +122,165 @@ pub fn infer_join_schema(
     Ok(Arc::new(schema))
 }
 
+struct JoinOutputColumn {
+    pub is_right: bool,
+    pub index: usize,
+    pub field: Field,
+}
+
+pub struct JoinOutputMapper {
+    mapping: Vec<JoinOutputColumn>,
```

Comment:
What about having two explicit mappings: `left_mapping` and `right_mapping`, then you don't need an `is_right` check.